### PR TITLE
Get rid of snapshot directory + related code cleanup and refactoring.

### PR DESCRIPTION
**Pull request by lubennikovaav**
- Add new subdir postgres_ffi/samples/ for config file samples. We only need pg_hba.conf. All other files can be empty.
- Don't copy wal to the new branch on zenith init or zenith branch.
- Import_timeline_wal on zenith init.
issue #325

Ready for review.
The diff looks a bit messy, but it is mostly removing unused code and rearranging what's left.

I need feedback on how to preserve pg_hba.conf sample.
@ericseppanen , does this code looks good to you:
`pub static PG_HBA: &'static str = include_str!("../samples/pg_hba.conf");`
Or is there a better way to include a sample into the package?

### CODE REVIEW

File: control_plane/src/compute.rs [hunk 1]
```diff
@@ -240,10 +241,19 @@ impl PostgresNode {
             },
         )?;
 
-        // FIXME: The compute node should be able to stream the WAL it needs from the WAL safekeepers or archive.
-        // But that's not implemented yet. For now, 'pg_wal' is included in the base backup tarball that
-        // we receive from the Page Server, so we don't need to create the empty 'pg_wal' directory here.
-        //fs::create_dir_all(pgdata.join("pg_wal"))?;
+        // Create pgdata subdirs structure
+        for i in 0..pg_constants::PGDATA_SUBDIRS.len() {
```
Comment (knizhnik) [inline control_plane/src/compute.rs:245]: Why do we need "i" here? Why just not 
```
for dir in pg_constants::PGDATA_SUBDIRS {
```

File: pageserver/src/basebackup.rs [hunk 2]
```diff
@@ -41,66 +41,41 @@ pub struct Basebackup<'a> {
 impl<'a> Basebackup<'a> {
     pub fn new(
         write: &'a mut dyn Write,
-        timelineid: ZTimelineId,
+        _timelineid: ZTimelineId,
```
Comment (knizhnik) [inline pageserver/src/basebackup.rs:44]: Should we just remove this unused field?

File: pageserver/src/basebackup.rs [hunk 3]
```diff
@@ -170,18 +147,41 @@ impl<'a> Basebackup<'a> {
...
-        info!("add_relmap_file {:?}", db);
         let path = if db.spcnode == pg_constants::GLOBALTABLESPACE_OID {
+            let dst_path = format!("PG_VERSION");
+            //TODO fix hardcoded value. Get this version num somewhere
+            let data = "14".as_bytes();
+            let header = new_tar_header(&dst_path, data.len() as u64)?;
+            self.ar.append(&header, &data[..])?;
+
+            let dst_path = format!("global/PG_VERSION");
+            let data = "14".as_bytes();
```
Comment (knizhnik) [inline pageserver/src/basebackup.rs:162]: Why not to use "data" defined above?

File: pageserver/src/basebackup.rs [hunk 4]
```diff
@@ -170,18 +147,41 @@ impl<'a> Basebackup<'a> {
...
             String::from("global/pg_filenode.map") // filenode map for global tablespace
         } else {
             // User defined tablespaces are not supported
             assert!(db.spcnode == pg_constants::DEFAULTTABLESPACE_OID);
-            let src_path = format!("{}/base/1/PG_VERSION", self.snappath);
+
+            // Append dir path for each database
+            let path = format!("base/{}", db.dbnode);
+
+            // FIXME It's a hack to send dir using append_dir()
```
Comment (knizhnik) [inline pageserver/src/basebackup.rs:174]: Is there more straightforward way to do it?
https://stackoverflow.com/questions/12844384/add-empty-directory-to-tarfile

**Review by knizhnik — CHANGES_REQUESTED** [review]

File: pageserver/src/basebackup.rs [hunk 5]
```diff
@@ -41,66 +41,41 @@ pub struct Basebackup<'a> {
...
-                    {
-                        trace!("sending {}", relpath.display());
-                        self.ar.append_path_with_name(fullpath, relpath)?;
-                    }
-                } else {
-                    // relation pages are loaded on demand and should not be included in tarball
-                    trace!("not sending {}", relpath.display());
-                }
+        // Send empty config files.
+        for i in 0..pg_constants::PGDATA_SPECIAL_FILES.len() {
```
Comment (hlinnaka) [inline pageserver/src/basebackup.rs:62]: This needs some more thinking.. Where should we grab the config files?

I think this is OK for now, but I opened an issue to track this https://github.com/zenithdb/zenith/issues/355

**Comment by hlinnaka** [comment]
Could extract the removal of 'identify_system' to a separate PR.

File: pageserver/src/basebackup.rs (hunk 4 above)
Comment (lubennikovaav) [reply pageserver/src/basebackup.rs:174]: Sadly this link is about python)

**Comment by lubennikovaav** [comment]
> Could extract the removal of 'identify_system' to a separate PR.
Done. Check #356

**Review by knizhnik — APPROVED** [review]

**Comment by lubennikovaav** [comment]
@LizardWizzard , please, review the rebase on top of multitenancy patch

File: pageserver/src/branches.rs [hunk 6]
```diff
@@ -119,66 +154,49 @@ pub fn create_repo(
...
-        conf,
-        std::sync::Arc::new(storage),
-        wal_redo_manager,
-        tenantid,
-    );
-    let timeline = repo.create_empty_timeline(tli, Lsn(lsn))?;
+    let lsn = get_lsn_from_controlfile(&pgdata_path)?;
 
-    restore_local_repo::import_timeline_from_postgres_datadir(&tmppath, &*timeline, Lsn(lsn))?;
+    println!("bootstrap_timeline {:?} at lsn {}", pgdata_path, lsn);
```
Comment (LizardWizzard) [inline pageserver/src/branches.rs:178]: probably should be an info! instead, because create repo can be called as a page server command

**Review by LizardWizzard — APPROVED** [review]
left a few nitpicks, looks good to me

**Pull request merged**